src/main/java/org/folio/rest/service/StreamService.java

```diff
@@ -66,16 +68,25 @@ public String enhanceFlux(String firstFluxId, Flux<String> secondFlux, String co
     while(firstString != null || secondString != null) {
       JsonNode firstObject = mapper.readTree(firstString);
       JsonNode secondObject = secondString != null ? mapper.readTree(secondString) : null;
-      JsonNode enhancementNode = secondObject != null ? secondObject.get(comparisonProperty) : null;
-      if (comparator.compare(firstString, secondString) == 0) {
+      JsonNode enhancementNode = secondObject != null ? secondObject.get(enhancementProperty) : null;
+      boolean matched = true;
+      for (Entry<String, String> entry : comparisonMap.entrySet()) {
+        Comparator<String> comparator = nullsLast(new PropertyComparator(entry.getKey(), entry.getValue()));
```

Comment:
The comparator should isolate the iteration of sort logic. This can be accomplished by having the comparator compare the `JsonNode` deserialized objects and passing the `comparisonMap` to the comparator constructor. This will prevent the double deserialization and make for more readable logic in this method.